gapis/api/gles/compat.go

```diff
@@ -562,6 +562,13 @@ func compat(ctx context.Context, device *device.Instance) (transform.Transformer
 				log.E(ctx, "Error decompressing texture: %v", err)
 			}
 
+		case *GlTexBufferEXT:
+			if !version.IsES && version.AtLeast(3, 1) { // Strip suffix on desktop.
```

Comment:
Well, we had this discussion before - flexibility vs simplicity.

You are going NxM approach where ideally, every trace can be replayed on any target, but every user can essentially have unique replay stream depending on his version and extensions.

I prefer more of 1:1 approach, where given trace always produces the same compat replay stream for every user, but we might potentially limit some users with old GL versions.

In this case, my approach with some old GL version will yield "Unsupported function glTexBuffer" with related atom number.  Not ideal, but not hard to understand either.  I would be in favor to add something long lines of "requiresGL(3,1)" which would log error, and record the fact that this trace requires GL3.1 and why.  We might even do the compat translation before choosing target device, so we can pick the replay device intelligently.

With has has been submitted, the compat will produce unmodified stream for this case.  Which might or might not work.  Worst case scenario, the driver supports the extension, but the extension is different to the non-extension and things behave differently.  No error is reported, but other things break.  There is no indication that GL3.1 is needed either.  The user is now executing code path that we have never tested, and that can not reproduce locally.  If we get the version and extension list, and stare at compat.go long enough, we might figure it out.  More likely we will just start closing bugs with "Works for me"